quinn-proto/src/crypto/rustls.rs

```diff
@@ -283,7 +284,9 @@ impl crypto::ServerConfig for rustls::ServerConfig {
         version: u32,
         params: &TransportParameters,
     ) -> Box<dyn crypto::Session> {
+        // TODO: why is this unwrap safe?
         let version = interpret_version(version).unwrap();
+        // TODO: why is this unwrap safe?
```

Comment:
Are these safe, and if so, why?

Reply:
`interpret_version` is safe per the docs on the declaration of `ServerConfig::start_session`:

> Never called if `initial_keys` rejected `version`.

The unwrap on `suite_from_provider` is not. Maybe we should validate it similarly?

Reply:
We now have derived `QuicClientConfig` and `QuicServerConfig` types that can be fallibly constructed from an existing rustls `ClientConfig` or `ServerConfig`, respectively.